src/box/iproto.cc

```diff
@@ -1656,11 +2011,27 @@ tx_process_call(struct cmsg *m)
 		goto error;
 	}
 
+	if (txn != NULL) {
+		/*
+		 * Transaction could be commited
+		 * or rollbacked in `call` or `eval`
+		 */
+		msg->stream->txn = in_txn();
+		fiber_set_txn(fiber(), NULL);
+	}
 	iproto_reply_select(out, &svp, msg->header.sync,
 			    ::schema_version, count);
 	iproto_wpos_create(&msg->wpos, out);
 	return;
 error:
+	if (txn != NULL) {
```

Comment:
That's a copy-paste from two lines above. It's better to create a function for that.

Reply:
Fixed